webknossos/script-collection/globalize_floodfill.py

```diff
@@ -0,0 +1,364 @@
+import os
+import re
+import time
+from argparse import ArgumentParser, Namespace
+from collections import namedtuple
+from glob import iglob
+from os import path
+from pathlib import Path
+from typing import Iterable, List, Optional, Set, Tuple
+
+import numpy as np
+import wkw
+
+from webknossos.geometry import BoundingBox, Mag, Vec3Int
+
+BUCKET_SIZE = 32
+BUCKET_SHAPE = (BUCKET_SIZE, BUCKET_SIZE, BUCKET_SIZE)
+CUBE_SHAPE = BUCKET_SHAPE  # (256, 256, 256)
+NEIGHBORS = [(1, 0, 0), (-1, 0, 0), (0, 1, 0), (0, -1, 0), (0, 0, 1), (0, 0, -1)]
+
+FloodFillBbox = namedtuple(
+    "FloodFillBbox",
+    ["bounding_box", "seed_position", "source_id", "target_id", "timestamp"],
+)
+
+
+def create_parser() -> ArgumentParser:
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "--volume_path",
+        "-v",
+        help="Directory containing the volume tracing.",
+        type=Path,
+    )
+
+    parser.add_argument(
+        "--segmentation_layer_path",
+        "-s",
+        help="Directory containing the segmentation layer.",
+        type=Path,
+    )
+
+    parser.add_argument(
+        "--nml_path",
+        "-n",
+        help="NML that contains the bounding boxes",
+        type=Path,
+    )
+
+    parser.add_argument("--output_path", "-o", help="Output directory", type=Path)
+
+    parser.add_argument("--skip_merge", default=False, action="store_true")
+
+    return parser
+
+
+def get_bucket_pos_and_offset(
+    global_position: Tuple[int, int, int]
+) -> Tuple[Tuple[int, int, int], Tuple[int, int, int]]:
+    offset = (
+        global_position[0] % CUBE_SHAPE[0],
+        global_position[1] % CUBE_SHAPE[1],
+        global_position[2] % CUBE_SHAPE[2],
+    )
+    return (
+        substract_positions(global_position, offset),
+        offset,
+    )
+
+
+def add_positions(
+    a: Tuple[int, int, int], b: Tuple[int, int, int]
+) -> Tuple[int, int, int]:
+    return a[0] + b[0], a[1] + b[1], a[2] + b[2]
+
+
+def substract_positions(
+    a: Tuple[int, int, int], b: Tuple[int, int, int]
+) -> Tuple[int, int, int]:
+    return a[0] - b[0], a[1] - b[1], a[2] - b[2]
```

Comment:
This can be replaced by our new `Vec3Int` abstraction which supports arithmetic operations :)